compiler/dyno/lib/parsing/ParserContextImpl.h

```diff
@@ -645,15 +656,18 @@ OpCall* ParserContext::buildUnaryOp(YYLTYPE location,
                                     PODUniqueString op,
                                     AstNode* expr) {
   auto ustrOp = UniqueString(op);
-
+  // may reassign op here to match old parser
+  // as in buildPreDecIncWarning
   if (ustrOp == "++") {
     noteWarning(location, "++ is not a pre-increment");
+    ustrOp = USTR("+");
   } else if (ustrOp == "--") {
-    noteWarning(location, "== is not a pre-decrement");
+    noteWarning(location, "-- is not a pre-decrement");
+    ustrOp = USTR("-");
```

Comment:
It's supposed to create nested calls, e.g. `--a` turns into `-(-a)`. You can still do this in this code you will just have to call an inner OpCall::build in this case, rather than just changing the op called. I recommend you make the `++` and `--` branches build what they need to and return it; so that the return below is only for the non-error cases.

Reply:
I wrapped these with another `OpCall` with the same operator, with the originally created `OpCall` node as the child node.